multiview/embed/kcca.py

```diff
@@ -0,0 +1,488 @@
+"""
+kcca.py
+====================================
+Python module for regularized kernel canonical correlation analysis.
+Code adopted from UC Berkeley, Gallant lab
+(https://github.com/gallantlab/pyrcca)
+"""
+
+import numpy as np
+from scipy.linalg import eigh
+
+
+class KCCA(object):
+    """
+    Kernel CCA class initialization and methods
+
+    Parameters
+    ----------
+    reg
+        Regularization parameter. Default is 0.1. (Float)
+    numCC
+        Number of canonical dimensions to keep. Default is 10. (Integer)
+    kernelcca
+        Kernel or non-kernel CCA. Default is True. (Boolean)
+    ktype
+        Type of kernel if kernelcca is True. (String)
+        Value can be 'linear' (default), 'gaussian' or 'polynomial'.
+    verbose
+        Provides detailed explanation of results. Default is True. (Boolean)
+    cutoff
+        Optional regularization parameter to perform spectral
+        cutoff when computing the canonical weight pseudoinverse
+        during held-out data prediction
+        Default is 1x10^-15 (Float)
+    gausigma
+        Parameter if the kernel is a Gaussian kernel. (Float)
+    degree
+        Parameter if the kernel is a Polynomial kernel. (Integer)
+
+    Returns
+    -------
+    ws_
+        Canonical weights (List)
+    comps_
+        Canonical components (List)
+    cancorrs_
+        Correlations of the canonical components on the training dataset (List)
+    corrs_
+        Correlations on the validation dataset (List)
+    preds_
+        Predictions on the validation dataset (List)
+    ev_
+        Explained variance for each canonical dimension (List)
+    """
+
+    def __init__(
+        self,
+        reg=None,
+        numCC=None,
+        kernelcca=True,
+        ktype=None,
+        verbose=False,
+        cutoff=1e-15,
+        gausigma=1.0,
+        degree=2,
+    ):
+        self.reg = reg
+        self.numCC = numCC
+        self.kernelcca = kernelcca
+        self.ktype = ktype
+        self.cutoff = cutoff
+        self.gausigma = gausigma
+        self.degree = degree
+        if self.kernelcca and self.ktype is None:
```

Comment:
should just make a default kwarg

Reply:
completed. changed it before you noted it.